tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot

```diff
@@ -68,29 +89,29 @@ Verify odh-model-controller Deployment
     Verify Deployment    ${odh_model_controller}  3  1  ${containerNames}
 
-Temporary Label MM Namespace
-    ${label} =    Run    oc label namespace ${MODEL_MESH_NAMESPACE} opendatahub.io/generated-namespace=true --overwrite=true
-    Log    ${label}
-    Run Keyword And Continue On Failure  Should Be Equal As Strings    ${label}    namespace/${MODEL_MESH_NAMESPACE} not labeled
-
 Verify Openvino Deployment
-    Run Keyword And Continue On Failure  Temporary Label MM Namespace
-    @{ovms} =  Oc Get    kind=Pod    namespace=${MODEL_MESH_NAMESPACE}    label_selector=name=modelmesh-serving-ovms-1.x
+    [Arguments]    ${project_name}=${PRJ_TITLE}    ${num_replicas}=1
+    @{ovms} =  Oc Get    kind=Pod    namespace=${project_name}   label_selector=name=modelmesh-serving-model-server-${project_name}
     ${containerNames} =  Create List  rest-proxy  oauth-proxy  ovms  ovms-adapter  mm
-    Verify Deployment    ${ovms}  2  5  ${containerNames}
-    ${all_ready} =    Run    oc get deployment -l name=modelmesh-serving-ovms-1.x | grep 2/2 -o
-    Should Be Equal As Strings    ${all_ready}    2/2
-
-Delete Model Serving Resources
-    [Documentation]    wrapper keyword that runs oc commands to delete model serving stuff
-    ...    Temporary until included in RHODS
-    ${output}=    Run    cd tests/Resources/Files && ./uninstall-model-serving.sh ${ODH_NAMESPACE} ${MODEL_MESH_NAMESPACE}
-    Log  ${output}
-    # Ignore all not found for delete commands, only output should be the not found of the patch command
-    Should Be Equal As Strings    Error from server (NotFound): kfdefs.kfdef.apps.kubeflow.org "odh-modelmesh" not found    ${output}
-
-Teardown Model Serving
-    [Documentation]  delete modelmesh stuff, Temporary until included in RHODS
-    # Seems like it's taking 10 minutes to stop reconciling deployments
-    Wait Until Keyword Succeeds  15 min  15s  Delete Model Serving Resources
-    Run    rm -rf modelmesh-serving
\ No newline at end of file
+    Verify Deployment    ${ovms}  ${num_replicas}  5  ${containerNames}
+    ${all_ready} =    Run    oc get deployment -n ${project_name} -l name=modelmesh-serving-model-server-${project_name} | grep ${num_replicas}/${num_replicas} -o
+    Should Be Equal As Strings    ${all_ready}    ${num_replicas}/${num_replicas}
+
+Get Access Token via UI
+    [Documentation]
+    [Arguments]    ${project_name}    ${service_account_name}=default-name
+    Open Data Science Projects Home Page
+    Project Should Be Listed    ${project_name}
+    Open Data Science Project Details Page    ${project_name}
+    ${token}=    Get Model Serving Access Token via UI    ${service_account_name}
+    Open Model Serving Home Page
+    [Return]    ${token}
+
+Model Serving Suite Teardown
+    [Documentation]    Suite teardown steps after testing DSG. It Deletes
+    ...                all the DS projects created by the tests and run RHOSi teardown
+    Delete Model Via UI    test-model
```

Comment:
instead of deleting models via UI in the teardown, would it be safer to delete it via CLI? Maybe you could have a separated test case to test the deletion via UI

Reply:
I think our main test should use the UI, since that will be what end users will also use, no?
We can have a CLI clean up in other tests where we don't explicitly care about this

Reply:
>I think our main test should use the UI


yeah, i was thinking of situations where this fails and the suite teardown fails to clean up. However, you are deleting the project via CLI so even if the first fails, the model should get deleted. Do you know if it is the case? Would the model be successfully deleted if the project gets deleted?

Reply:
Yes, it would

Reply:
ok then!